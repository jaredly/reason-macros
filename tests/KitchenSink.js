// Generated by BUCKLESCRIPT VERSION 4.0.1100, PLEASE EDIT WITH CARE
'use strict';


console.log(2);

/*  Not a pure module */
