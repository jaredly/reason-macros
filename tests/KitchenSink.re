
let%macro sink = (cap: capIdent, longCap: longCapIdent, id: ident, longId: longIdent, pat: pattern, exp: expression, num: int, isTrue: bool) => {
  let eval__id = 2;
  let%eval newVar = 5;
  let m = Js.log(eval__id);
  let m = if%eval (isTrue) {
    eval__newVar;
  } else {
    eval__newVar * 20;
  };
  if%eval (num - 14 > 20) {
    Js.log("Large number: $eval{num}")
  }
  switch (eval__exp) {
    | Eval__cap(47) => None
    | eval__pat => eval__exp
    | _ => None
  };
  let%eval (myMap: map) = {one: 2, three: "four", someIdent: "yes"};
  let%eval number = [%eval get(myMap, "$eval{id}")];
  eval__number
};

let%macro dev = (expression: expression) => if%eval (env("node_env") != "production") { eval__expression } else { () };

let%macro env = (vbl: string) => [%eval env(vbl)];

[%sink (Some, Some.Long.Cap, someIdent, Some.Long.ident, [%pat? Some(_)], None, 35, false)]

let%macro.pat apply = (ident: string, args: pattern) => [%pat? Pexp_apply({pexp_desc: Pexp_ident({txt: Lident(eval__ident)})}, eval__args)];

switch expr {
  | [%apply ("++", [%pat? [(_, one), (_, two)]])] => one
  | _ => ()
};

[%dev print_endline("Hello world")];

let editor = [%env "EDITOR"];