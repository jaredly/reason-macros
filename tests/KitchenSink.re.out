{
  let someIdent = 2;
  let m = Js.log(someIdent);
  switch (None) {
  | Some(47) => None

  | Some(_) => None
  | _ => None
  };
};
